core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -56,43 +58,88 @@
 import java.util.concurrent.TimeUnit;
 
 import javax.annotation.Nullable;
+import javax.net.ssl.SSLSession;
 
 /**
  * Common utilities for GRPC.
  */
+// TODO(lukaszx0) replace all deprecated calls internally and remove deprecated fields in 1.0 (?)
 public final class GrpcUtil {
 
   /**
    * {@link io.grpc.Metadata.Key} for the timeout header.
    */
-  public static final Metadata.Key<Long> TIMEOUT_KEY =
+  public static final Metadata.Key<Long> TIMEOUT_METADATA_KEY =
```

Comment:
Since I added new type of key to this util class - `Attributes.Key` - to avoid confusion I refactored metadata keys and added `_METADATA_`  in the name of existing `_KEY` fields to easily distinguish them and also be able to accommodate other types of keys in the future.

It's a public, widely used API in apps so it warrants a special handling, hence `@Deprecated`. I intentionally didn't change all the call sites in core cause it'd make this PR unreadable with a lot of unrelated refactorings - I'll send a follow up cleaning this up.